### DISCUSSION TITLE
camera facing billboards

### DISCUSSION THREAD

**Discussion by HawkenKing** [google/model-viewer#4692] [Ideas]
![r1](https://github.com/google/model-viewer/assets/5999766/ddce6066-ed21-4d76-b943-3718f81ecdd1)

A commonly used effect in games and other 3D production is the ability to mark certain objects with a flag that makes them point at the camera. This can be used to great effect but if often overlooked by online model viewers like sketchfab etc.

https://threejs.org/examples/?q=bill#webgl_points_billboards

This would make for a great addition to model-viewer! If we could mark an object by hierarchy name and it's Z+ would then face the camera at all times, for example:

-model
--house
--cactus-MVbillboard
--cactus-MVbillboardYlock

The "-MVbillboard" part would be interpreted from the glTF as an object that needs to face the camera.
Ylock is a technique to only make the Z+ face the cameras position as if it were on the same vertical plane (as with these cacti)

Without Ylock it would face the camera like a sprite.

[edit] some more links on object3D.lookAt() :

https://github.com/blairmacintyre/aframe-look-at-billboard-component

https://stackoverflow.com/questions/59954885/aframe-billboarding-only-y-axis

https://threejs.org/docs/#api/en/core/Object3D.lookAt

**Comment by elalish** [comment]
This kind of thing sounds like a purview of the KHR_interactivity extension, which we do plan to support once the spec is stable. Meanwhile, you can also do this via HTML using our annotations, though it won't have 3D occlusion. https://modelviewer.dev/examples/annotations/index.html

**Reply by HawkenKing** [reply]
going to try this on some simple billboards, thanks!
